### DISCUSSION TITLE
Global content for entire collection with placeholders

### DISCUSSION THREAD

**Discussion by carstenjaksch** [statamic/cms#11237] [Q&A] [answered]
I have a collection for landingpages and a mounted entry for that collection.

The mounted entry contains the global content inside a replicator page builder.

The collection entries contain several fields to replace content in specific areas (text, images and sometimes arrays of data). Additionally, the title of a collection entry is often used in the content of the mounted entry's page builder fields.

I've already created a template for the collection that loads the page builder content from the mounted entry:

```
<main class="stack-12 md:stack-16 lg:stack-24" id="content">
    {{ if {mount_url:{collection}} }}
        {{ get_content from="{{mount_url:{collection}}}" }}
            {{ page_builder scope="block" }}
                {{ partial src="page_builder/{type}" }}
            {{ /page_builder }}
        {{ /get_content }}
    {{ /if }}
</main>
```

My plan is to use `{{ page:field_name }}` to access the fields of the currently viewed collection entry, regardless of the context (inside the mounted page builder or even deeper), replace placeholders like `%title%` and call it a day.

But I have some concerns:

- I would need to use the `replace` tag very often in partials that are also rendered in "normal" entries. Would that be a critical performance hit?
- As I understand, the `replace` partial can only replace one placeholder at once. Is there an elegant way of replacing multiple placeholders with one tag? Do I need to create a custom one? Or is there an elegant way of placeholder handling?
- Is that even a "good" practice what I do? Or is there a better way to do such things?

I have this same setup in WordPress, Elementor has a feature to load dynamic data into a field. Pretty neat, the HTML output is garbage though. Maybe there is an alternative way of doing so with Antlers in a field (I have noticed the option to enable that). But there is also a warning about security and performance.

**Comment by edalzell** [comment]
I don't understand the "replacing" bit, can you please show an example of what you mean? Pseudo-code is fine.

**Reply by carstenjaksch** [reply]
For example, there might be a title field like "Try our services in %title%" and in the template I would do `{{ title | replace('%title%', page:title) }}` to have something like "Try our services in London".

Maybe enabled Antlers for a field would be more efficient. But I need to do that with images too, where Antlers is not possible.

I hope that part is easier to understand now.

**Reply by edalzell** [reply]
why not just use the page content? it'll override the mounted page content?

**Reply by carstenjaksch** [reply]
I don't want to override all of it, just parts. Or am I not understanding you right now?

**Reply by edalzell** [reply, marked as answer]
Statamic uses "the cascade" for data, if it's local (i.e. the current page data) it'll use that, if it's in the cascade (like the `get_content` stuff) it'll use that.

Override what you want.

**Reply by carstenjaksch** [reply]
Thank you for mentioning the cascade. I had a quick read-up and it is a very nice feature – helps me a lot with partials (I don't need to explicitly pass params) and I could replace images and other fields if they are missing.

For text fields, I think enabled Antlers tags in fields are my best bet. Seems to work, like Shortcodes in WordPress.

Any downsides to Antlers in fields? Security – okay, I have to live with that. Performance – aggressive caching could help? The content of these entries rarely change.

**Reply by duncanmcclean** [reply]
> Any downsides to Antlers in fields? Security – okay, I have to live with that. Performance – aggressive caching could help? The content of these entries rarely change.

If you know what you're doing (eg. you're not letting strangers edit your content, pointing the clients to learn Antlers), then I don't think there's any problem in doing it. I do the same on my own website. 🙂

**Reply by carstenjaksch** [reply]
Thank you both for your quick responses. Helped me a lot!
